frame/support/src/storage/unhashed.rs

```diff
@@ -154,6 +154,13 @@ pub fn clear_prefix(
 	MultiRemovalResults { maybe_cursor, backend: i, unique: i, loops: i }
 }
 
+pub fn contains_prefix(prefix: &[u8]) -> bool {
```

Comment:
Looks like in storage maps the keys are encoded before hashing so even with identity hasher we would have no empty keys.

So this would just need a different name and doc to make sure user do not expect corner cases to work (corner cases being `contains_prefix([])` on db containing only `[]` and contains `contains_prefix(b"prefix")` on db containing only b"prefix" as key starting with b"prefix") . Or just implement corner cases.

Reply:
Personally, I'd just warn the user about unexpected behavior for `[]`.
I don't see the point of implementing edge cases because I can't think about a use-case for them, and it would just become additional complexity for a function which aims to be as lightweight as possible.
Anyway, I'll wait for some feedback before doing it.

Reply:
My suggestion was about `frame/support/src/storage/unhashed.rs' contains_prefix function, here the function prefix is exposed and can be misused (not through storage map api).

Storage map with empty keys do not really make sense indeed.

Imagine looking at storage unhashed, seeing `contains_prefix` and using it just to realize that it does not work if prefix contains only `prefix` key (tbh I would probably read the code (if called `is_empty_prefix` I may trust the function name)).

Reply:
Sorry for the misunderstanding. 
I'd rather comment something like 
> Returns `true` if any key in the storage starts with said prefix, including full matches.